hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmMultipartUpload.java

```diff
@@ -41,32 +41,13 @@ public class OmMultipartUpload {
 
   private ReplicationConfig replicationConfig;
 
-  public OmMultipartUpload(String volumeName, String bucketName,
-      String keyName, String uploadId) {
-    this.volumeName = volumeName;
-    this.bucketName = bucketName;
-    this.keyName = keyName;
-    this.uploadId = uploadId;
-  }
```

Comment:
This constructor sets the "DB key" part of `OmMultipartUpload`.  It is used by `KeyManagerImpl.listMultipartUploads` (via `from()`), which looks up the object by the DB key, then fills the rest of the info (creation time, replication config).  Other code expects these additional data to be present in `OmMultipartUpload`, so they are not optional, but the incomplete constructor does not enforce this.

https://github.com/apache/ozone/blob/d2ac33627ad83e8a3e5e8ec6db113a9d4779612b/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java#L619-L640

I think it would be better to extract these fields of `OmMultipartUpload` (plus the constructor and `from()`) into a separate class (`OmMultipartUploadDbKey`?).  `from()` would create an instance of the DB key instead of the incomplete `OmMultipartUpload`.  This would allow `OmMultipartUpload` to always be completely initialized (with `OmMultipartUploadDbKey`, creation time and replication config), and only one constructor would be left in `OmMultipartUpload`.